### ISSUE TITLE
Mobile app 

### ISSUE THREAD

**Issue by antonyukden50** [miroslavpejic85/mirotalk#16]
Can I connect from the mobile application (IOS, ANDROID ) to the browser room?

**Comment by miroslavpejic85** [comment]
Hi @antonyukden50, welcome!
Yes of course, just open the `browser` on your mobile device and paste in the room link.
For example: https://mirotalk.herokuapp.com/join/antonyukden50
Ref: https://github.com/miroslavpejic85/mirotalk#room-name

**Comment by antonyukden50** [comment]
It is clear) 
But meant whether in the mobile application ios and android it would be possible to integrate 
Was it possible to contact the browser from the mobile application (crossplatform)

**Comment by miroslavpejic85** [comment]
A ok, you mean create an app mirotalk for Android - iOS?
The idea was to `not depend` on any `app or anything else` just save the link on your phone, that can be for example: 
direct access to your room: https://mirotalk.herokuapp.com/join/antonyukden50
or 
Create the room name: https://mirotalk.herokuapp.com/newcall
And then just share it with the first left button in the room, that it's also optimized for the mobile.
Why you need an app, if everything works through a `browser` mobile - desktop? 🙂

**Comment by Jaideep25** [comment]
Hey @antonyukden50 !
If you want to make it into an app, use any one of these!
`AppsGeyser` : https://appsgeyser.com/
`Thunkable` : https://thunkable.com

**Comment by miroslavpejic85** [comment]
You could just include it in a frame in the app ;)
``` html
<iframe allow="camera; microphone; fullscreen; display-capture; autoplay" src="https://mirotalk.herokuapp.com/newcall" style="height: 100%; width: 100%; border: 0px;"></iframe>
```

**Comment by Jaideep25** [comment]
Wow cool!!

**Comment by anshuman2166AppproDev** [comment]
I made an android distro for mieotalk, lemmey drop it's source

**Comment by anshuman2166AppproDev** [comment]
It is easy to make mobile app by using Java Webview class in android studio

**Comment by miroslavpejic85** [comment]
Yes, Implement `mirotalk` in `app` (ios - android), if some one need,  I think it can be done in a few lines of code, as I said for example by including it in a `iframe`.

**Comment by anshuman2166AppproDev** [comment]
> Yes, Implement `mirotalk` in `app` (ios - android), if someone need, I think it can be done in a few lines of code, as I said for example by including it in a `iframe`.
> If I see many requests, I will implement it in the future.

It is easier than u even think Use MIT APPINVENTOR and just drag their webviewer component and enter url (Mit appinventor helps you build apps with drag n drop)

**Comment by Jaideep25** [comment]
> > Yes, Implement `mirotalk` in `app` (ios - android), if someone need, I think it can be done in a few lines of code, as I said for example by including it in a `iframe`.
> > If I see many requests, I will implement it in the future.
> 
> It is easier than u even think Use MIT APPINVENTOR and just drag their webviewer component and enter url (Mit appinventor helps you build apps with drag n drop)

Actually the same is in Thunkable too !

**Comment by Nisarga-Developer** [comment]
> Hey @antonyukden50 !
> If you want to make it into an app, use any one of these!
> `AppsGeyser` : https://appsgeyser.com/
> `Thunkable` : https://thunkable.com


Unfortunately, Thunkable X & Appsgeyser doesn't support mic & cam access

**Comment by Jaideep25** [comment]
> > Hey @antonyukden50 !
> > If you want to make it into an app, use any one of these!
> > `AppsGeyser` : https://appsgeyser.com/
> > `Thunkable` : https://thunkable.com
> 
> Unfortunately, Thunkable X & Appsgeyser doesn't support mic & cam access

Did you try it?

**Comment by anshuman2166AppproDev** [comment]
> > Hey @antonyukden50 !
> > If you want to make it into an app, use any one of these!
> > `AppsGeyser` : https://appsgeyser.com/
> > `Thunkable` : https://thunkable.com
> 
> Unfortunately, Thunkable X & Appsgeyser doesn't support mic & cam access

Ufff, mit appinventor supports by usng vknow sir;s extension

**Comment by Nisarga-Developer** [comment]
Thunkable's not App Inventor. In order to make phone compatible the only way is to use Android Studio or use ai2 with customwebview (github.com/vknow360/customwebview) or use Webview Permission Handler extension by Niotron store. Make sure you grant android.permission.MICROPHONE and android.permission.CAMERA

**Comment by Nemra1** [comment]
> Yes, Implement `mirotalk` in `app` (ios - android), if someone need, I think it can be done in a few lines of code, as I said for example by including it in a `iframe`.
> If I see many requests, I will implement it in the future.

i have a plane to convert it into cordova  will be awesome

**Comment by ugale-deepak3010** [comment]
> > Yes, Implement `mirotalk` in `app` (ios - android), if someone need, I think it can be done in a few lines of code, as I said for example by including it in a `iframe`.
> > If I see many requests, I will implement it in the future.
> 
> i have a plane to convert it into cordova will be awesome

actually you are comming in real point.

**Comment by ugale-deepak3010** [comment]
If we want to use in Android other Cordova that is required this app convert to server side php
And client side node. Js

I think signalling and other portion is required to convert into that ways.

But most one more problem is mobile device on can not share screen.

**Comment by miroslavpejic85** [comment]
You're talking about this one: https://cordova.apache.org/
Regarding mobile screen sharing at the now is not supported, only desktop.

**Comment by yogo1212** [comment]
Sorry for nectrobumping :-)

When using mirotalk in the browser on a smartphone, the call freezes when the tab goes into the background.

Is there a way to interface with a Mirotalk server through an API?

**Comment by nottheend** [comment]
@yogo1212 agree. unfortunately, as of now the very good mirotalk is not convenient to use on Android. As soon as the browser is long enough in background or the phone locks itself, the conference gets unusable.
I am wondering if there is not a generic webrtc client for android out there, which could be used?
